components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -345,7 +367,7 @@ tobeDeleted claims (claims came from federated idp as null). If there is a match
     }
 
     protected void associateUser(String username, String userStoreDomain, String tenantDomain, String subject,
-                                 String idp) throws FrameworkException {
+                                 String idp, boolean isFederatedUser) throws FrameworkException {
```

Comment:
Overload the protected method. Additionally if the old method should not be used deprecate it.